lib/datasets.go

```diff
@@ -625,3 +625,54 @@ func (r *DatasetRequests) ManifestMissing(a, b *dag.Manifest) (err error) {
 	*b = *mf
 	return
 }
+
+// DAGInfo generates a dag.Info for a dataset path
+func (r *DatasetRequests) DAGInfo(refstr *string, i *dag.Info) (err error) {
+	if r.cli != nil {
+		return r.cli.Call("DatasetRequests.DAGInfo", refstr, i)
+	}
+
+	ref, err := repo.ParseDatasetRef(*refstr)
+	if err != nil {
+		return err
+	}
+	if err = repo.CanonicalizeDatasetRef(r.node.Repo, &ref); err != nil {
+		return
+	}
+
+	var info *dag.Info
+	info, err = actions.NewDAGInfo(r.node, ref.Path)
+	if err != nil {
+		return
+	}
+	*i = *info
+	return
+}
+
+// SubDAGParams defines parameters for the SubDAGInfo method
+type SubDAGParams struct {
+	RefStr, Label string
+}
+
+// SubDAGInfo generates a dag.Info for a sub DAG at a label in the original dag.Info
+func (r *DatasetRequests) SubDAGInfo(s *SubDAGParams, i *dag.Info) error {
```

Comment:
hrmmm... possible to have the main `DAGInfo` command accept what is currently `SubDAGParams` and if the "Label" field is empty just give the main `DAGInfo`? it would remove the need for this method entirely

Reply:
yea I thought about this, and actually started doing it that way at first. Then I figure it would be easier to condense it once I got it working separately. Was hoping for this exact feedback.

Reply:
totally. I find myself writing separate implementations & re-combining them after I understand both all the time. I think it's a healthy pattern especially given that if it's not caught on the first pass, subsequent refactoring usually uncovers this same conclusion.